File: Selenium/Activities/Activity8.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import io.netty.util.Timeout;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.io.IOException;
import java.time.Duration;

public class Activity8 {
    public static void main(String[] args) throws IOException {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/dynamic-controls");
        System.out.println("Title of webpage:"+driver.getTitle());

        WebDriverWait wait=new WebDriverWait(driver,Duration.ofSeconds(10));

        WebElement toggleCheckbox=driver.findElement(By.id("toggleCheckbox"));
        WebElement checkBox=driver.findElement(By.id("dynamicCheckbox"));

        toggleCheckbox.click();
        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.id("dynamicCheckbox")));
        System.out.println("Whether checkbox present on webpage:"+checkBox.isDisplayed());

        toggleCheckbox.click();
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("dynamicCheckbox")));
        System.out.println("Whether checkbox present on webpage:"+checkBox.isDisplayed());

        driver.findElement(By.xpath("//input[@class='willDisappear']")).click();
        driver.close();

    }
}

File: Selenium/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    driver.get("https://v1.training-support.net/selenium/dynamic-controls")
    wait = WebDriverWait(driver, 10)
    print("Title of page:" + driver.title)
    toggleCheckbox = driver.find_element(By.ID, "toggleCheckbox")
    checkBox = driver.find_element(By.ID, "dynamicCheckbox")
    toggleCheckbox.click()
    wait.until(expected_conditions.invisibility_of_element(toggleCheckbox))
    print("Whether checkbox present on webpage:" + str(checkBox.is_displayed()))
    toggleCheckbox.click()
    print("Whether checkbox present on webpage:" + str(checkBox.is_displayed()))
    driver.find_element(By.XPATH, "//input[@class='willDisappear'")
    driver.close()
